coremltools/models/neural_network/builder.py

```diff
@@ -3346,7 +3346,12 @@ def check_valid_preprocessing_keys(input, target, input_name):
                             output_name=input_transpose
                         )
                         layers = self.nn_spec.layers
-                        layers.insert(0, layers.pop())
```

Comment:
Can we add a simple unit test to make sure this particular change doesn't break again?

Reply:
I'm not sure how to add a simple unit test, if there are some code I can refer to it would be nice.